src/core_write.cpp

```diff
@@ -107,7 +107,8 @@ std::string ScriptToAsmStr(const CScript& script, const bool fAttemptSighashDeco
             str += "[error]";
             return str;
         }
-        if (0 <= opcode && opcode <= OP_PUSHDATA4) {
+        static_assert(std::numeric_limits<decltype(opcode)>::min() == 0);
```

Comment:
Like, yes it is, but I wanted a 1 for 1 for the runtime check just done at compile time; they will evaluate the same for all normal integer types; idk 🤷‍♂️

Reply:
It's constexpr since C++17, so, should work for any type:
```
template< class T >
inline constexpr bool is_unsigned_v = is_unsigned<T>::value;
```
which elsee runtime checks do you mean?

Reply:
Yeah, for sure it's constexpr, but the runtime check I'm removing is specifically `0 <= opcode` and the most intuitive way imo to say that's always true is to say the min is 0; but honestly using the unsigned is probably appropriate too

Reply:
Actually interesting it seems that `static_assert(std::is_unsigned_v<decltype(opcode)>);` fails as opcodetype is an enum and I guess enums aren't considered unsigned?